tests/rptest/tests/datalake/delayed_translation_test.py

```diff
@@ -0,0 +1,238 @@
+                             self._partition_count(),
+                             config={
+                                 "redpanda.remote.read": "true",
+                                 "redpanda.remote.write": "true",
+                                 "retention.local.target.bytes": 1024,
+                                 "retention.local.target.ms": 100,
+                                 "cleanup.policy": "delete",
+                             })
+            rpk.create_schema_from_str(self.SCHEMA_NAME, self.AVRO_SCHEMA_STR)
+
+            self._connect = RedpandaConnectService(self.test_context,
+                                                   self.redpanda)
+            self._connect.start()
+
+            # produce until all partitions are GCed locally
+
+            local_starts = []
+
+            def update_local_starts_and_check_truncated():
+                nonlocal local_starts
+                local_starts = self._for_each_partition(
+                    self._partition_local_starts)
+                self.redpanda.logger.debug(f"{local_starts=}")
+                return all(local_start > 0 for local_start in local_starts)
+
+            self.produce_until(
+                update_local_starts_and_check_truncated,
+                timeout_sec=600,
+                backoff_sec=1,
+                err_msg="Failed waiting for all partitions to be GCed")
+
+            # enable iceberg
+
+            self.redpanda.set_cluster_config(
+                {
+                    "iceberg_enabled": True,
+                    "iceberg_catalog_commit_interval_ms": 5000,
+                    "iceberg_target_lag_ms": 5000
+                },
+                expect_restart=True)
+
+            for k, v in {
+                    TopicSpec.PROPERTY_ICEBERG_MODE: "key_value",
+                    TopicSpec.PROPERTY_ICEBERG_TARGET_LAG_MS: 10000
+            }.items():
+                rpk.alter_topic_config(self.TOPIC_NAME, k, v)
+
+            # check translation started for all partitions
+
+            iceberg_max_offsets = []
+
+            def all_partitions_get_translated():
+                try:
+                    self._query_minmax(dl.spark())
+                except Exception as e:
+                    self.redpanda.logger.info(f"Error querying: {e}")
+                    return False
+
+                for pid, query_result in self._last_query_result.items():
+                    assert query_result["min_offset"] >= local_starts[pid]
+
+                self.redpanda.logger.info(
+                    f"{len(self._last_query_result)} of "
+                    f"{self._partition_count()} partitions translated")
+                return len(self._last_query_result) == self._partition_count()
+
+            wait_until(
+                all_partitions_get_translated,
+                timeout_sec=300,
+                backoff_sec=1,
+                err_msg="Failed waiting for all partitions to be translated")
```

Comment:
when running locally for 1000 partitions it often fails here as some partitions translation is stuck catching unexpected errors here https://github.com/redpanda-data/redpanda/blob/6dc1950bc71096b80a6eae792c2055630602f595/src/v/datalake/translation/partition_translator.cc#L281-L288